File: activity/20200805/scau-flyingbird/Simplify_Path.go
package main

import "strings"

func simplifyPath(path string) string {
	strPath := strings.Split(path, "/")

	var simpPath []string

	for _, str := range strPath {
		if strings.EqualFold(str, "") {
			continue
		} else if strings.EqualFold(str, ".") {
			continue
		} else if strings.EqualFold(str, "..") {
			if len(simpPath) > 0 {
				simpPath = simpPath[:len(simpPath)-1]
			}
		} else {
			simpPath = append(simpPath, str)
		}
	}

	return "/" + strings.Join(simpPath, "/")

	return ""
}

File: activity/20200805/RabbitLee/Simplify_Path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        res = list()
        
        # split input
        directories = [d for d in path.split('/') if d]
        
        # process directories
        for d in directories:
            if d == '.':
                pass
            elif d == '..':
                if res:
                    res.pop()
            else:
                res.append(d)
                
        # combine to canonical path
        return '/' + '/'.join(res)

